list/list.go

```diff
@@ -0,0 +1,203 @@
+	} `json:"spec"`
+}
+
+func ListPolicies(c *k8s.Client, o Options) error {
+	env, err := CheckEnv(c)
+	if err != nil {
+		return err
+	}
+	if env == "k8s" {
+		nsList, err := c.K8sClientset.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		policyList := v1.KubeArmorPolicyList{}
+		for _, namespace := range nsList.Items {
+			policies, err := c.KSPClientset.KubeArmorPolicies(namespace.Name).List(context.Background(), metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			policyList.Items = append(policyList.Items, policies.Items...)
+		}
+		switch format := o.Output; format {
+		case "json":
+			result, err := json.MarshalIndent(policyList.Items, "", "")
+			if err != nil {
+				return err
+			}
+			fmt.Println(string(result))
+			return nil
+		default:
+			err = printK8SPolicies(policyList.Items)
+			if err != nil {
+				return err
+			}
+
+		}
+	} else if env == "systemd" {
+		policies, err := getSystemdPolicies()
+		if err != nil {
+			return err
+		}
+		switch format := o.Output; format {
+		case "json":
+			result, err := json.Marshal(policies)
+			if err != nil {
+				return err
+			}
+			fmt.Println(string(result))
+			return nil
+		default:
+			err := printSystemdPolices(policies)
+			if err != nil {
+				return err
+			}
+		}
+
+	}
+	return nil
+}
+
+func printK8SPolicies(policies []v1.KubeArmorPolicy) error {
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Policy Name", "Namespace", "Pods"})
+	data := [][]string{}
+	if len(policies) <= 0 {
+		fmt.Println("No policies found in your cluster")
+		return nil
+	}
+	for _, policy := range policies {
+		name := policy.ObjectMeta.Name
+		ns := policy.ObjectMeta.Namespace
+		pods := policy.Spec.Selector.MatchLabels["container"]
+		data = append(data, []string{name, ns, pods})
+	}
+
+	for _, pol := range data {
+		table.Append(pol)
+	}
+	table.Render()
+	return nil
+}
+
+func printSystemdPolices(policies []HostPolicyBackup) error {
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Policy Name", "Host"})
+	data := [][]string{}
+	for _, policy := range policies {
+		name := policy.Metadata.PolicyName
+		host := policy.Spec.NodeSelector.MatchLabels.KubernetesIoHostname
+		data = append(data, []string{name, host})
+	}
+	for _, pol := range data {
+		table.Append(pol)
+	}
+	table.Render()
+	return nil
+}
+
+func getSystemdPolicies() ([]HostPolicyBackup, error) {
+	files, err := ioutil.ReadDir("/opt/kubearmor/policies")
+	if err != nil {
+		if errors.Is(err, fs.ErrPermission) {
+			return nil, errors.New("unable to read policy backups. Try running with elevated priviledges")
+		}
+		return []HostPolicyBackup{}, err
+	}
+	if len(files) <= 0 {
+		fmt.Printf("Unable find host polices")
+		return nil, err
+	}
+	policies := []HostPolicyBackup{}
+	for _, file := range files {
+		path := filepath.Join("/opt/kubearmor/policies", filepath.Clean(file.Name()))
```

Comment:
@Ankurk99  the go sec error seems to be caused by this line , however I am already cleaning the file paths

Reply:
I think it's because we are trying to access the path which otherwise would not be possible by a non-root user.

Reply:
oh ,  I guess this is ok? wdyt?